common/events/events_io.cpp

```diff
@@ -123,6 +123,32 @@ void AsyncWriterFromWriter::Cancel() {
 	}
 }
 
+ReaderFromAsyncReader::ReaderFromAsyncReader(AsyncReaderFromEventLoopFunc func) {
+	reader_ = func(event_loop_);
+}
+
+mio::ExpectedSize ReaderFromAsyncReader::Read(
+	vector<uint8_t>::iterator start, vector<uint8_t>::iterator end) {
+	size_t read;
+	error::Error err;
```

Comment:
You're right, I even got confused when I went back now to read it. Changed it to use a dedicated `inner_err` variable which is now captured instead of the `err`.